test/jasmine/tests/transform_multi_test.js

```diff
@@ -446,6 +439,88 @@ describe('multiple transforms:', function() {
         });
     });
 
+    it('executes filter and aggregate in the order given', function() {
+        // filter and aggregate do not commute!
+
+        var trace1 = {
+            x: [0, -5, 7, 4, 5],
+            y: [2, 4, 6, 8, 10],
+            transforms: [{
+                type: 'aggregate',
+                groups: [1, 2, 2, 1, 1],
+                aggregations: [
+                    {array: 'x', func: 'sum'},
+                    {array: 'y', func: 'avg'}
+                ]
+            }, {
+                type: 'filter',
+                target: 'x',
+                operation: '<',
+                value: 5
+            }]
+        };
+
+        var trace2 = Lib.extendDeep({}, trace1);
+        trace2.transforms.reverse();
+
+        Plotly.newPlot(gd, [trace1, trace2]);
+
+        var trace1Out = gd._fullData[0];
+        expect(trace1Out.x).toEqual([2]);
+        expect(trace1Out.y).toEqual([5]);
+
+        var trace2Out = gd._fullData[1];
+        expect(trace2Out.x).toEqual([4, -5]);
+        expect(trace2Out.y).toEqual([5, 4]);
+    });
+
+    it('always executes groupby before aggregate', function() {
+        // aggregate and groupby wouldn't commute, but groupby always happens first
+        // because it has a `transform`, and aggregate has a `calcTransform`
```

Comment:
You mean the other way around - should it be possible to aggregate and then group? Yes of course, there are definitely use cases for it - say you want to do a `'count'` aggregation, then group by that count - one group for all entries with one sample, a separate group for entries with two samples, etc etc. But it seems like for the moment we have a deeper structural problem with that, so this test documents it.